lib/plausible/auth/user.ex

```diff
@@ -259,11 +259,13 @@ defmodule Plausible.Auth.User do
     end
   end
 
-  defp set_email_verified(user) do
-    if Keyword.fetch!(Application.get_env(:plausible, :selfhost), :enable_email_verification) do
+  defp change_email_verification_status(user) do
+    on_full_build do
       change(user, email_verified: false)
     else
-      change(user, email_verified: true)
+      selfhosted_config = Application.get_env(:plausible, :selfhost)
+      enable? = Keyword.fetch!(selfhosted_config, :enable_email_verification)
+      change(user, email_verified: !enable?)
```

Comment:
`ENABLE_EMAIL_VERIFICATION` env var is casted to an (existing) atom which is expected to be either `true` or `false`, so I think we can rephrase that as:

```suggestion
      must_verify? = Keyword.fetch!(selfhosted_config, :enable_email_verification)
      change(user, email_verified: not must_verify?)
```

WDYT?